Army/ArmyTest.h

```diff
@@ -0,0 +1,80 @@
+#include <cxxtest/TestSuite.h>
+
+#include "../units/unit.h"
+#include "../units/soldier.h"
+#include "../units/rogue.h"
+#include "../units/berserk.h"
+#include "../units/vampire.h"
+#include "../units/werewolf.h"
+#include "../state/state.h"
+#include "../state/wolfstate.h"
+#include "../state/humanstate.h"
+
+#include "../mystic/spellcaster.h"
+#include "../mystic/wizard.h"
+    
+#include "../magicskill/spell.h"
+
+
+using namespace std;
+
+class ArmyTest : public CxxTest::TestSuite {
+    public:
+
+        void testSoldier() {
+            Soldier* soldier = new Soldier("Steve", 100, 10);
+
+            soldier->heal(150);
```

Comment:
Why only healing?
I'd use soldier for all basic unit functionality. All other classes then can be tested only for specific functionality